src/vm/jitinterface.cpp

```diff
@@ -4503,6 +4503,63 @@ CorInfoType CEEInfo::getTypeForPrimitiveValueClass(
     return result;
 }
 
+/*********************************************************************/
+CorInfoType CEEInfo::getTypeForPrimitiveNumericClass(
+        CORINFO_CLASS_HANDLE clsHnd)
+{
+    CONTRACTL {
+        SO_TOLERANT;
+        THROWS;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+    } CONTRACTL_END;
+
+    CorInfoType result = CORINFO_TYPE_UNDEF;
+
+    JIT_TO_EE_TRANSITION();
```

Comment:
This can be `..._LEAF`.

Reply:
Thank you! I am confused by `JIT_TO_EE_TRANSITION` and `JIT_TO_EE_TRANSITION_LEAF`. How can I decide to use which one?

Reply:
You need to look at the operations done within the implementation. If they do not throw or load new types, it can use the cheaper "LEAF" transition.